### PR TITLE
[AUTOMATE-2866] Remove v1 users APIs from gateway

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description: What code changed, and why?

Also cleaned up some of the client usage in `automate-cli`

### :athletic_shoe: How to Build and Test the Change

```
rebuild components/authz-service && rebuild components/automate-gateway && rebuild components/automate-cli
# delete various things related to the admin (user, remove from team, delete the team, remove team from policy etc.)
chef-automate iam admin-access restore chefautomate
```

### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### Same failures at feature branch

https://buildkite.com/chef-oss/chef-automate-master-verify/builds/9499#31846972-246b-495c-aa7a-52920c7d93a9
https://buildkite.com/chef/chef-automate-master-verify-private/builds/9796#0550852d-bef9-40f2-87f4-ea7b049c06f2

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/iam.go [hunk 1]
```diff
@@ -199,52 +198,18 @@ func runRestoreDefaultAdminAccessAdminCmd(cmd *cobra.Command, args []string) err
...
-		}
-
-		writer.Skipped("Found local team: admins in Chef-managed policy: Admin")
-	default:
-		// do nothing
+	if !foundAdminsTeaminV2AdminPolicy {
+		writer.Success("Added local team: admins to Chef-managed policy: Admin")
 	}
 
+	writer.Skipped("Found local team: admins in Chef-managed policy: Admin")
```
Comment (msorens) [inline components/automate-cli/cmd/chef-automate/iam.go]: While you are in here how about make these flow better (and correct name):
```suggestion
	if !foundAdminsTeaminV2AdminPolicy {
		writer.Success("Added local 'admins' team to Chef-managed 'Administrator' policy")
	}

	writer.Skipped("Found local 'admins' team in Chef-managed 'Administrator' policy")
```

File: components/automate-cli/pkg/adminmgmt/adminmgmt.go [hunk 2]
```diff
@@ -8,15 +8,9 @@ import (
...
-	authz_constants_v1 "github.com/chef/automate/components/authz-service/constants/v1"
 	authz_constants_v2 "github.com/chef/automate/components/authz-service/constants/v2"
 	"github.com/chef/automate/components/automate-cli/pkg/client"
 	"github.com/chef/automate/components/automate-cli/pkg/status"
-	teams "github.com/chef/automate/components/automate-gateway/api/auth/teams"
-	teams_req "github.com/chef/automate/components/automate-gateway/api/auth/teams/request"
-	users_req "github.com/chef/automate/components/automate-gateway/api/auth/users/request"
-	"github.com/chef/automate/components/automate-gateway/api/authz"
-	authz_req "github.com/chef/automate/components/automate-gateway/api/authz/request"
 	authz_v2_req "github.com/chef/automate/components/automate-gateway/api/iam/v2/request"
```
Comment (msorens) [inline components/automate-cli/pkg/adminmgmt/adminmgmt.go]: Per previous team discussion, let's try to minimize v2 references where possible. So this should become:
```suggestion
	authz_req "github.com/chef/automate/components/automate-gateway/api/iam/v2/request"
```

File: components/automate-cli/pkg/adminmgmt/adminmgmt.go [hunk 3]
```diff
@@ -123,49 +115,6 @@ func AddUserToTeam(ctx context.Context,
...
-			}
-		}
-	}
-
-	return foundOriginalTeamsPolicy, createdNewPolicy, nil
-}
-
 // UpdateV2AdminsPolicyIfNeeded fetches the chef-managed Admin policy's members
 // and adds the admins team if it's missing from that list
 func UpdateV2AdminsPolicyIfNeeded(ctx context.Context,
```
Comment (msorens) [inline components/automate-cli/pkg/adminmgmt/adminmgmt.go]: Similarly:
```suggestion
func UpdateAdminPolicyIfNeeded(ctx context.Context,
```

File: components/automate-cli/pkg/adminmgmt/adminmgmt_test.go [hunk 4]
```diff
@@ -136,96 +138,70 @@ func TestEnsureTeam(t *testing.T) {
...
 
-			id, found, err := adminmgmt.CreateAdminTeamIfMissing(ctx, apiClient, false)
+			found, err := adminmgmt.CreateAdminTeamIfMissing(ctx, apiClient, false)
 			require.NoError(t, err)
-			assert.Equal(t, createAdminID, id)
 			assert.False(t, found)
 		})
 
-		t.Run("and dry run mode is on it is not created, id is empty, and found=false are returned", func(t *testing.T) {
+		t.Run("and dry run mode is on it is not created and found=false is returned", func(t *testing.T) {
```
Comment (msorens) [inline components/automate-cli/pkg/adminmgmt/adminmgmt_test.go]:
```suggestion
		t.Run("and dry run mode is on, it is not created and found=false is returned", func(t *testing.T) {
```

File: components/automate-cli/pkg/adminmgmt/adminmgmt_test.go [hunk 5]
```diff
@@ -335,113 +309,6 @@ func TestAddAdminUserToTeam(t *testing.T) {
...
-		serverMocks.AuthzMock.CreatePolicyFunc = createDefaultAdminPolicy
-
-		foundPolicy, createdNewPolicy, err := adminmgmt.UpdateV1AdminsPolicyIfNeeded(ctx, apiClient, false)
-		require.NoError(t, err)
-		assert.True(t, foundPolicy)
-		assert.True(t, createdNewPolicy)
-	})
-}
-
 func TestUpdateV2AdminsPolicyIfNeeded(t *testing.T) {
```
Comment (msorens) [inline components/automate-cli/pkg/adminmgmt/adminmgmt_test.go]: Match suggested function name change mentioned earlier:
```suggestion
func TestUpdateAdminPolicyIfNeeded(t *testing.T) {
```

File: components/automate-cli/pkg/adminmgmt/adminmgmt_test.go [hunk 6]
```diff
@@ -11,15 +11,11 @@ import (
...
-	authz_constants_v1 "github.com/chef/automate/components/authz-service/constants/v1"
 	"github.com/chef/automate/components/automate-cli/pkg/adminmgmt"
 	"github.com/chef/automate/components/automate-cli/pkg/client/mock"
-	teams_req "github.com/chef/automate/components/automate-gateway/api/auth/teams/request"
-	teams_resp "github.com/chef/automate/components/automate-gateway/api/auth/teams/response"
-	users_req "github.com/chef/automate/components/automate-gateway/api/auth/users/request"
-	users_resp "github.com/chef/automate/components/automate-gateway/api/auth/users/response"
 	authz_req "github.com/chef/automate/components/automate-gateway/api/authz/request"
 	authz_resp "github.com/chef/automate/components/automate-gateway/api/authz/response"
+	policies_common "github.com/chef/automate/components/automate-gateway/api/iam/v2/common"
```
Comment (bcmdarroch) [inline components/automate-cli/pkg/adminmgmt/adminmgmt_test.go]:
```suggestion
	iam_common "github.com/chef/automate/components/automate-gateway/api/iam/v2/common"
```
(and same with the req and resp imports below)

File: components/automate-cli/pkg/client/mock/mock.go [hunk 7]
```diff
@@ -7,9 +7,6 @@ import (
 
 	"github.com/chef/automate/api/external/applications"
 	"github.com/chef/automate/components/automate-cli/pkg/client"
-	"github.com/chef/automate/components/automate-gateway/api/auth/teams"
-	"github.com/chef/automate/components/automate-gateway/api/auth/tokens"
-	"github.com/chef/automate/components/automate-gateway/api/auth/users"
 	"github.com/chef/automate/components/automate-gateway/api/authz"
 	"github.com/chef/automate/components/automate-gateway/api/compliance/reporting"
 	v2 "github.com/chef/automate/components/automate-gateway/api/iam/v2"
```
Comment (bcmdarroch) [inline components/automate-cli/pkg/client/mock/mock.go]:
```suggestion
	iam "github.com/chef/automate/components/automate-gateway/api/iam/v2"
```

File: components/automate-cli/pkg/client/apiclient/apiclient.go [hunk 8]
```diff
@@ -9,9 +9,6 @@ import (
 	client_type "github.com/chef/automate/components/automate-cli/pkg/client"
 	"github.com/chef/automate/components/automate-cli/pkg/status"
 	"github.com/chef/automate/components/automate-deployment/pkg/constants"
-	"github.com/chef/automate/components/automate-gateway/api/auth/teams"
-	"github.com/chef/automate/components/automate-gateway/api/auth/tokens"
-	"github.com/chef/automate/components/automate-gateway/api/auth/users"
 	"github.com/chef/automate/components/automate-gateway/api/authz"
 	"github.com/chef/automate/components/automate-gateway/api/compliance/reporting"
 	v2 "github.com/chef/automate/components/automate-gateway/api/iam/v2"
```
Comment (bcmdarroch) [inline components/automate-cli/pkg/client/apiclient/apiclient.go]:
```suggestion
	iam "github.com/chef/automate/components/automate-gateway/api/iam/v2"
```

File: components/automate-cli/pkg/client/types.go [hunk 9]
```diff
@@ -2,9 +2,6 @@ package client
 
 import (
 	"github.com/chef/automate/api/external/applications"
-	"github.com/chef/automate/components/automate-gateway/api/auth/teams"
-	"github.com/chef/automate/components/automate-gateway/api/auth/tokens"
-	"github.com/chef/automate/components/automate-gateway/api/auth/users"
 	"github.com/chef/automate/components/automate-gateway/api/authz"
 	"github.com/chef/automate/components/automate-gateway/api/compliance/reporting"
 	v2 "github.com/chef/automate/components/automate-gateway/api/iam/v2"
```
Comment (bcmdarroch) [inline components/automate-cli/pkg/client/types.go]:
```suggestion
	iam "github.com/chef/automate/components/automate-gateway/api/iam/v2"
```

File: components/automate-ui/src/app/components/landing/landing.component.spec.ts [hunk 10]
```diff
@@ -23,9 +23,9 @@ describe('LandingComponent', () => {
       anyOfCheck: [['/auth/something-else', 'get', '']],
       route: '/settings/teams'
     },
-    { allOfCheck: [['/auth/users', 'get', '']], route: '/settings/users' },
+    { allOfCheck: [['/iam/v2/users', 'get', '']], route: '/settings/users' },
     {
-      anyOfCheck: [['/auth/tokens', 'get', ''], ['/iam/v2/tokens', 'get', '']],
+      anyOfCheck: [['/iam/v2/tokens', 'get', ''], ['/iam/v2/tokens', 'get', '']],
```
Comment (bcmdarroch) [inline components/automate-ui/src/app/components/landing/landing.component.spec.ts]: I think this is covered in @blakestier's PR? Also it's repeating the same path

Comment (blakestier) [reply components/automate-ui/src/app/components/landing/landing.component.spec.ts]: This is in my PR (and is repeating the same path)

File: components/automate-cli/cmd/chef-automate/iam.go [hunk 11]
```diff
@@ -183,11 +181,12 @@ func runRestoreDefaultAdminAccessAdminCmd(cmd *cobra.Command, args []string) err
...
 	// In dry-run mode, we might be missing some IDs that would have been created.
 	// We'll only hit this condition in dry-run mode.
-	if iamCmdFlags.dryRun && (userID == "" || adminsTeamID == "") {
+	if iamCmdFlags.dryRun && (membershipID == "" || !adminsTeamFound) {
 		writer.Success("Added admin user to admins team")
 	} else { // non-dry-run mode or dry-run mode where user and team already existed.
 		userAdded, err := adminmgmt.AddAdminUserToTeam(ctx,
-			apiClient, adminsTeamID, userID, iamCmdFlags.dryRun)
+			// TODO
+			apiClient, "admins", membershipID, iamCmdFlags.dryRun)
```
[the remark this answers is not in the document]
Comment (blakestier) [reply components/automate-cli/cmd/chef-automate/iam.go]: We probably want to use a constant there

File: components/automate-cli/cmd/chef-automate/dev.go [hunk 12]
```diff
@@ -994,23 +994,23 @@ func runCreateIAMDevUsersCmd(*cobra.Command, []string) error {
...
 	for username, data := range map[string]struct {
-		displayName, password, team string
+		displayName, password, teamID string
 	}{
 		"viewer": {"Viewer User", "chefautomate", "viewers"},
 		"editor": {"Editor User", "chefautomate", "editors"},
 	} {
 		userID, _, err := adminmgmt.CreateUserOrUpdatePassword(ctx,
-			apiClient, username, data.displayName, data.password, false /* dry run */)
+			apiClient, username, data.displayName, data.password, false)
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/automate-cli/cmd/chef-automate/dev.go]: I disagree that this comment is useful. The function signature says what the boolean is doing. We don't have similar comments everywhere else we pass a boolean.

File: components/automate-cli/cmd/chef-automate/dev.go [hunk 13]
```diff
@@ -994,23 +994,23 @@ func runCreateIAMDevUsersCmd(*cobra.Command, []string) error {
...
 		if err != nil {
 			return err
 		}
 		// Note: the teams SHOULD exist. But since you never know what happens in a
 		// long running acceptance env, we'll better ensure them:
-		teamID, _, err := adminmgmt.EnsureTeam(ctx, data.team, data.team /* description */, apiClient, false /* dry run */)
+		_, err = adminmgmt.EnsureTeam(ctx, data.teamID, data.teamID, apiClient, false)
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/automate-cli/cmd/chef-automate/dev.go]: At least one integration test. I didn't look into it too heavily besides just not breaking it:

```
|Tylers-MacBook-Pro automate|++> git grep create-iam-dev-users
.studiorc:  chef-automate dev create-iam-dev-users
components/automate-cli/cmd/chef-automate/dev.go:               Use:   "create-iam-dev-users",
integration/tests/iam_v2.sh:    chef-automate dev create-iam-dev-users || return 1
terraform/test-environments/modules/chef_automate_install/templates/install_chef_automate_cli.sh.tpl:chef-automate dev create-iam-dev-users
```

File: components/automate-cli/cmd/chef-automate/iam.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/automate-cli/cmd/chef-automate/iam.go]: Yep! Lemme fix that.

**Pull request merged**